d2l-icon.html

```diff
@@ -0,0 +1,75 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-icon/iron-icon.html">
+<link rel="import" href="../d2l-colors/d2l-colors.html">
+<link rel="import" href="../d2l-polymer-behaviors/d2l-dom.html">
+<dom-module id="d2l-icon">
+	<template>
+		<style include="d2l-colors">
+			:host {
+				color: var(--d2l-color-tungsten);
+				--iron-icon-height: 18px;
+				--iron-icon-width: 18px;
+			}
+			:host([icon*="d2l-tier2:"]) {
+				--iron-icon-height: 24px;
+				--iron-icon-width: 24px;
+			}
+			:host([icon*="d2l-tier3:"]) {
+				--iron-icon-height: 30px;
+				--iron-icon-width: 30px;
+			}
+			:host-context([dir="rtl"]) iron-icon {
+				transform: scaleX(-1);
+			}
+			:host([no-rtl]) iron-icon {
+				transform: none;
+			}
+		</style>
+		<iron-icon icon="[[icon]]"></iron-icon>
+	</template>
+	<script>
+		Polymer({
+			is: 'd2l-icon',
+			properties: {
+				icon: String,
+				noRtl: {
+					type: Boolean,
+					readOnly: true,
+					reflectToAttribute: true,
+					value: false
+				}
+			},
+			ready: function() {
+				var isNoRtl = this.__isNoRtl();
+				this._setNoRtl(isNoRtl);
+			},
```

Comment:
Aside from that, if we see the need for this sort of recursive function, it could be factored out similar to what I had done with [findComposedAncestor](https://github.com/Brightspace/d2l-polymer-behaviors-ui/blob/master/d2l-dom.html#L7).  In such a case, this could could become:

```
var svg = D2L.Dom.findComposedDescendant(
   node,
   function(node) {
      return (node.tagName.toLowerCase === 'svg');
   }
); 
```
Just thinking out loud...